vsutil/clips.py

```diff
@@ -229,3 +237,21 @@ def _should_dither(in_bits: int,
                 or float_to_int
                 or (in_range == types.Range.FULL and upsampling and (in_bits, out_bits) != (8, 16))
                 or downsampling)
+
+
+def shift_clip(clip: vs.VideoNode, offset: int) -> vs.VideoNode:
+    if offset > clip.num_frames - 1:
+        raise ValueError("shift_clip: Offset can't be greater than the clip length!")
+
+    if offset < 0:
+        return clip[0] * abs(offset) + clip[:offset]
+    elif offset > 0:
+        return clip[offset:] + clip[-1] * offset
+
+    return clip
+
+
+def shift_clip_multi(clip, shift: types.FrameRange = (-1, 1)) -> List[vs.VideoNode]:
+    ranges = func.normalize_franges(shift)
+
+    return [shift_clip(clip, x) for x in ranges]
```

Comment:
Can you give a real-world example for how this would be used, specifically the `_multi` variant?

Reply:
Single:
<https://github.com/Setsugennoao/stgfunc/blob/master/stgfunc/mask.py#L162>
Multi:
<https://github.com/Setsugennoao/stgfunc/blob/master/stgfunc/tweaking.py#L292>
<https://github.com/Setsugennoao/Encoding-Scripts/blob/master/Other%20Groups/gsk-kun/Eiga%20Daisuki%20Pompo-san%20(BD)/pompo_san.vpy#L221-L222>